storage/spec/acls_spec.rb

```diff
@@ -0,0 +1,435 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative "../acls"
+require "rspec"
+require "google/cloud/storage"
+require "tempfile"
+
+describe "Google Cloud Storage ACL sample" do
+
+  before do
+    @bucket_name     = ENV["GOOGLE_CLOUD_STORAGE_BUCKET"]
+    @storage         = Google::Cloud::Storage.new
+    @project_id      = @storage.project
+    @bucket          = @storage.bucket @bucket_name
+    @local_file_path = File.expand_path "resources/file.txt", __dir__
+    @test_email      = "user-test@test.com"
+  end
+
+  # Delete given file in Cloud Storage test bucket if it exists
+  def delete_file storage_file_path
+    @bucket.file(storage_file_path).delete if @bucket.file storage_file_path
+  end
+
+  def upload local_file_path, storage_file_path
+    unless @bucket.file storage_file_path
+      @bucket.create_file local_file_path, storage_file_path
+    end
+  end
+
+  def email_in_bucket_acl? email
+    @bucket.acl.owners.include?(email)    ||
+      @bucket.acl.writers.include?(email) ||
+      @bucket.acl.readers.include?(email)
+  end
+
+  def email_in_default_bucket_acl? email
+    @bucket.default_acl.owners.include?(email)    ||
+      @bucket.default_acl.readers.include?(email)
+  end
+
+  def email_in_file_acl? file_path, email
+    file = @bucket.file file_path
+
+    file.acl.owners.include?(email)    ||
+      file.acl.readers.include?(email)
+  end
+
+  # Capture and return STDOUT output by block
+  def capture &block
+    real_stdout = $stdout
+    $stdout = StringIO.new
+    block.call
+    @captured_output = $stdout.string
+  ensure
+    $stdout = real_stdout
+  end
+  attr_reader :captured_output
+
+  it "can print owner bucket acl" do
+    @bucket.acl.delete(@test_email) if email_in_bucket_acl?(@test_email)
+    @bucket.acl.reload!
+
+    expect(@bucket.acl.owners).not_to include(@test_email)
+
+    @bucket.acl.add_owner @test_email
+    @bucket.reload!
+
+    expect(@bucket.acl.owners).to include(@test_email)
+
+    capture do
+      print_bucket_acl project_id: @project_id, bucket_name: @bucket_name
+    end
+
+    expect(captured_output).to include "OWNER #{@test_email}"
+  end
+
+  it "can print writer bucket acl" do
+    @bucket.acl.delete(@test_email) if email_in_bucket_acl?(@test_email)
+    @bucket.acl.reload!
+
+    expect(@bucket.acl.writers).not_to include(@test_email)
+
+    @bucket.acl.add_writer @test_email
+    @bucket.acl.reload!
+
+    expect(@bucket.acl.writers).to include(@test_email)
+
+    capture do
+      print_bucket_acl project_id: @project_id, bucket_name: @bucket_name
```

Comment:
Under pretty much every other circumstance in the world, I would say let's keep them.

For code sample snippet tests, I believe the cost of maintenance and the cost of the extra LOC at the top of each of these tests outweighs the benefit.  It is preferable to have 1 example test case per snippet.  That makes it easier to navigate and understand the tests.

I love the thought.  And, in application code, all 3 branches of tests would be required.  But I say delete them in our case and have 1:1 snippet:test